jsonschema-validation.md

```diff
@@ -320,115 +309,57 @@ can be used alongside the `type` keyword with a value of "integer", or could be
-  annotation collection;
-- MUST evaluate `format` as an assertion;
-- MUST implement syntactic validation for all format attributes defined in this
-  specification, and for any additional format attributes that it recognizes,
-  such that there exist possible instance values of the correct type that will
-  fail validation.
-
 The requirement for minimal validation of format attributes is
 intentionally vague and permissive, due to the complexity involved in many of
 the attributes. Note in particular that the requirement is limited to syntactic
 checking; it is not to be expected that an implementation would send an email,
 attempt to connect to a URL, or otherwise check the existence of an entity
-identified by a format instance.[^4]
-
-[^4]: The expectation is that for simple formats such as date-time, syntactic
-validation will be thorough. For a complex format such as email addresses, which
-are the amalgamation of various standards and numerous adjustments over time,
-with obscure and/or obsolete rules that may or may not be restricted by other
-applications making use of the value, a minimal validation is sufficient. For
-example, an instance string that does not contain an "@" is clearly not a valid
-email address, and an "email" or "hostname" containing characters outside of
-7-bit ASCII is likewise clearly invalid.
-
-It is RECOMMENDED that implementations use a common parsing library for each
-format, or a well-known regular expression. Implementations SHOULD clearly
-document how and to what degree each format attribute is validated.
-
-The [standard core and validation meta-schema](#meta-schema) includes this
-vocabulary in its `$vocabulary` keyword with a value of false, since by default
-implementations are not required to support this keyword as an assertion.
-Supporting the format vocabulary with a value of true is understood to greatly
-increase code size and in some cases execution time, and will not be appropriate
-for all implementations.
+identified by a format instance.
 
 #### Custom format attributes
 
 Implementations MAY support custom format attributes. Save for agreement between
 parties, schema authors SHALL NOT expect a peer implementation to support such
-custom format attributes. An implementation MUST NOT fail to collect unknown
-formats as annotations. When the Format-Assertion vocabulary is specified,
-implementations MUST fail upon encountering unknown formats.
+custom format attributes.
 
-Vocabularies do not support specifically declaring different value sets for
-keywords.  Due to this limitation, and the historically uneven implementation of
-this keyword, it is RECOMMENDED to define additional keywords in a custom
-vocabulary rather than additional format attributes if interoperability is
-desired.
+An implementation MUST NOT fail to collect unknown formats as annotations.
+
+When configured for assertion behavior for `format`, implementations MUST fail
+upon encountering unknown formats.
```

Comment:
We say that even when assertion behavior is enabled, implementations might still just return `true`, so why must an implementation fail when encountering an unknown format rather than returning `true`? Or should they fail if they don't have a best-effort implementation for the format whether it's known or not?

I think this requirement made sense applied to the `format-assertion` vocabulary, but doesn't really work in this context.

Reply:
The answer to the question is: which is more acceptable, a false negative or a false positive?  I landed on the side of false negative being more acceptable.

Reply:
I wasn't advocating for any particular behavior. I was just pointing out that the text is contradictory.

But, my opinion on the behavior is that in assert mode it should either:

- Allow partial/no validation and unknown formats return true; OR
- Require full validation and refuse to process unknown formats

Allowing no validation and refusing to process unknown formats doesn't make sense.

Reply:
Do we just remove assertion completely, then, and let implementations figure out what that mode means?

Reply:
As long as we define something before we release, I'm fine with leaving this somewhat undefined for now.

Reply:
I think I want to leave this for #1520 to figure out.  For now, I just want to pull out the vocabulary language.  We'll definitely address it before release, though.